homeassistant/components/shelly/diagnostics.py

```diff
@@ -18,26 +18,35 @@ async def async_get_config_entry_diagnostics(
     """Return diagnostics for a config entry."""
     data: dict = hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry.entry_id]
 
+    device_settings: str | dict = "not initialized"
+    device_info: str | dict = "not initialized"
```

Comment:
Device info is available also if device is initialized so we don't need to protect it